### ISSUE TITLE
Sequences with "N" bases are not supported

### ISSUE THREAD

**Issue by jzhang-dev** [fawaz-dabbaghieh/bubble_gun#2]
In `BubbleGun/functions.py`, the `reverse_complement` function does not support "N" bases:

```py
def reverse_complement(dna):
    complement = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A'}
    return ''.join([complement[base] for base in dna[::-1]])
```

This will likely cause errors when processing input sequences with "N" bases. 

```
Traceback (most recent call last):
  ...
  File ".../BubbleGun/functions.py", line 11, in <listcomp>
    return ''.join([complement[base] for base in dna[::-1]])
                    ~~~~~~~~~~^^^^^^
KeyError: 'N'
```

A solution is to assume the complement base for "N" is still "N", i.e.:

```py
def reverse_complement(dna):
    complement = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A', 'N': 'N'}
    return ''.join([complement[base] for base in dna[::-1]])
```

**Comment by fawaz-dabbaghieh** [comment]
True! I haven't really worked with graphs with Ns, so I forgot to include that. Thanks for pointing this out. Should be now fixed with commit [7ec8027](https://github.com/fawaz-dabbaghieh/bubble_gun/commit/7ec8027c52c2f6104b345064ea158366459801fd)

**Comment by jzhang-dev** [comment]
@fawaz-dabbaghieh Thanks for working on this very helpful tool!
